presto-main/src/main/java/com/facebook/presto/execution/scheduler/nodeSelection/TopologyAwareNodeSelector.java

```diff
@@ -138,9 +140,11 @@ public SplitPlacementResult computeAssignments(Set<Split> splits, List<RemoteTas
 
         // todo identify if sorting will cause bottleneck
         List<HostAddress> sortedCandidates = sortedNodes(nodeMap);
+        NodeProvider<HostAddress> nodeProvider = new ModularHashingNodeProvider<>(sortedCandidates);
```

Comment:
- We also need a switch/case here, don't we?
- move this part of logic to `getNodeProvider` as suggested in another comment